app/src/main/java/com/lagradost/quicknovel/providers/RanobesProvider.kt

```diff
@@ -0,0 +1,264 @@
+import kotlin.random.Random
+
+
+val mapper = jacksonObjectMapper().apply {
+    propertyNamingStrategy = PropertyNamingStrategy.LOWER_CAMEL_CASE
+    setSerializationInclusion(JsonInclude.Include.NON_NULL)
+}
+
+data class Chapterdatajson (
+    @get:JsonProperty("book_title")
+    val bookTitle: String? = null,
+
+    @get:JsonProperty("book_link")
+    val bookLink: String? = null,
+
+    @get:JsonProperty("book_id")
+    val bookID: Long? = null,
+
+    val chapters: List<Chapter>? = null,
+
+    @get:JsonProperty("pages_count")
+    val pagesCount: Long? = null,
+
+    @get:JsonProperty("count_all")
+    val countAll: Long? = null,
+    val cstart: Long? = null,
+    val limit: Long? = null,
+    val search: String? = null,
+    val default: List<Any?>? = null,
+    val searchTimeout: Any? = null
+) {
+    fun toJson() = mapper.writeValueAsString(this)
+
+    companion object {
+        fun fromJson(json: String) = mapper.readValue<Chapterdatajson>(json)
+    }
+}
+
+data class Chapter (
+    val id: String? = null,
+    val title: String? = null,
+    val date: String? = null,
+    val showDate: String? = null,
+    val link: String? = null
+)
+
+
+class RanobesProvider : MainAPI() {
+    override val name = "Ranobes"
+    override val mainUrl = "https://ranobes.net"
+    override val hasMainPage = true
+
+    override val iconId = R.drawable.icon_ranobes
+
+    override val iconBackgroundId = R.color.white
+
+    override val tags = listOf(
+        Pair("Action", "Action"),
+        Pair("Adult", "Adult"),
+        Pair("Adventure", "Adventure"),
+        Pair("Comedy", "Comedy"),
+        Pair("Drama", "Drama"),
+        Pair("Ecchi", "Ecchi"),
+        Pair("Fantasy", "Fantasy"),
+        Pair("Game", "Game"),
+        Pair("Gender Bender", "Gender%20Bender"),
+        Pair("Harem", "Harem"),
+        Pair("Josei", "Josei"),
+        Pair("Historical", "Historical"),
+        Pair("Horror", "Horror"),
+        Pair("Martial Arts", "Martial%20Arts"),
+        Pair("Mature", "Mature"),
+        Pair("Mecha", "Mecha"),
+        Pair("Mystery", "Mystery"),
+        Pair("Psychological", "Psychological"),
+        Pair("Romance", "Romance"),
+        Pair("School Life", "School%20Life"),
+        Pair("Sci-fi", "Sci-fi"),
+        Pair("Seinen", "Seinen"),
+        Pair("Shoujo", "Shoujo"),
+        Pair("Shounen", "Shounen"),
+        Pair("Slice of Life", "Slice%20of%20Life"),
+        Pair("Shounen Ai", "Shounen%20Ai"),
+        Pair("Sports", "Sports"),
+        Pair("Supernatural", "Supernatural"),
+        Pair("Smut", "Smut"),
+        Pair("Tragedy", "Tragedy"),
+        Pair("Xianxia", "Xianxia"),
+        Pair("Xuanhuan", "Xuanhuan"),
+        Pair("Wuxia", "Wuxia"),
+        Pair("Yaoi", "Yaoi"),
+    )
+
+    override fun loadMainPage(
+        page: Int,
+        mainCategory: String?,
+        orderBy: String?,
+        tag: String?
+    ): HeadMainPageResponse {
+        val url = "$mainUrl/tags/genre/$tag/"
```

Comment:
This was the bug in the previus providers you merged, you have to take into account the page number by adding "page/$page/"